src/master-playlist-controller.js

```diff
@@ -1168,19 +1167,34 @@ export class MasterPlaylistController extends videojs.EventTarget {
     if (!mainStartingMedia || (hasAltAudio && !this.audioSegmentLoader_.startingMedia_)) {
       return;
     }
-    const audioStartingMedia = this.audioSegmentLoader_ && this.audioSegmentLoader_.startingMedia_;
+    const audioStartingMedia = this.audioSegmentLoader_ && this.audioSegmentLoader_.startingMedia_ || {};
     const media = this.masterPlaylistLoader_.media();
     const playlistCodecs = codecsForPlaylist(this.masterPlaylistLoader_.master, media);
     const codecs = {};
 
-    if (mainStartingMedia.hasAudio || hasAltAudio) {
-      codecs.audio = mainStartingMedia.audioCodec || (audioStartingMedia || {}).audioCodec ||
-        translateLegacyCodec(playlistCodecs.audio) || DEFAULT_AUDIO_CODEC;
-    }
+    // priority of codecs: playlist -> mux.js parsed codecs -> default
+    if (mainStartingMedia.isMuxed) {
```

Comment:
For muxed we mix the video and audio codec into one source buffer as a "video" codec